### ISSUE TITLE
Enhancement: [rule-name] <a short description of my proposal>

### ISSUE THREAD

**Issue by wolverineks** [typescript-eslint/typescript-eslint#6623]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/strict-boolean-expressions/

### Description

can this rule check for nullable string literals?
something like `"allowNullableStringLiteral": true`

### Fail

```typescript
function shouldError (foo: string | undefined) {
  if (!foo) {
    console.log(foo) // could be undefined or ""
  }
}
```

### Pass

```typescript
function shouldPass (foo: "A" | "B" | undefined) {
  if (!foo) {
    console.log(foo) // can only be undefined
  }
}
```

**Comment by bradzacher** [comment]
This is working as intended.

The rule checks for cases where you have multiple truthy or multiple falsey cases.

`'A' | undefined` has exactly one truthy case and one falsey case, which is why it is not flagged by the rule.

I don't believe an option for this would be useful to the broader community because the options are intended to relax the rule, not make it stricter. Additionally from experience - the vast majority of codebases don't want a check like this.

**Comment by bradzacher** [comment]
Please also take the time to fill out the issue title next time.

**Comment by wolverineks** [comment]
I meant if the options are set to disallow string comparisons, then there is no way to allow literal string comparison, even if empty string is not part of the union type.
So, enforce checking for empty strings with `someString === ""` instead of `!someString`
but allow checking for existence of nullable string literal with  `!nonEmptyString`
